src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ObjectOrCollectionInitializerExpression.cs

```diff
@@ -442,19 +441,125 @@ private BoundExpression VisitObjectInitializerMember(BoundObjectInitializerMembe
                         if (!isRhsNestedInitializer)
                         {
                             // Rewrite as simple assignment.
-                            var rewrittenRight = VisitExpression(assignment.Right);
+                            var rewrittenRight = VisitExpression(right);
                             result.Add(MakeStaticAssignmentOperator(assignment.Syntax, rewrittenAccess, rewrittenRight, false, assignment.Type, used: false));
                             return;
                         }
 
                         break;
                     }
 
+                case BoundKind.ImplicitIndexerAccess:
+                    if (temps == null)
+                    {
+                        temps = ArrayBuilder<LocalSymbol>.GetInstance();
+                    }
+
+                    var implicitIndexer = (BoundImplicitIndexerAccess)left;
+
+                    BoundExpression transformedImplicitIndexer;
+                    var argumentType = implicitIndexer.Argument.Type;
+                    if (TypeSymbol.Equals(argumentType, _compilation.GetWellKnownType(WellKnownType.System_Range), TypeCompareKind.ConsiderEverything))
+                    {
+                        transformedImplicitIndexer = VisitRangePatternIndexerAccess(implicitIndexer, temps, result, inInitializer: true);
+
+                        Debug.Assert(transformedImplicitIndexer is BoundCall
+                        {
+                            Arguments: [BoundPassByCopy { Expression: BoundLocal { LocalSymbol: SynthesizedLocal } },
+                                        BoundPassByCopy { Expression: BoundLocal { LocalSymbol: SynthesizedLocal } }]
+                        });
+                    }
+                    else
+                    {
+                        transformedImplicitIndexer = TransformImplicitIndexerAccess(implicitIndexer,
+                            isRegularAssignmentOrRegularCompoundAssignment: true, inInitializer: true, result, temps, isDynamicAssignment: false);
+
+                        Debug.Assert(transformedImplicitIndexer is BoundArrayAccess { Indices: [BoundLocal { LocalSymbol: SynthesizedLocal }] }
+                            or BoundIndexerAccess { Arguments: [BoundLiteral or BoundLocal { LocalSymbol: SynthesizedLocal }] }
+                            or BoundLocal { LocalSymbol: SynthesizedLocal });
+                    }
+
+                    if (transformedImplicitIndexer is BoundIndexerAccess indexerAccess)
```

Comment:
>if (transformedImplicitIndexer is BoundIndexerAccess indexerAccess) [](http://example.com/codeflow?start=20&length=67)

Consider inlining this logic into the previous ```if```. It looks like only one branch needs to worry about this case.